sdkproject/Assets/Mapbox/Core/Unity/MeshGeneration/Factories/TerrainFactory.cs

```diff
@@ -43,6 +43,12 @@ public class TerrainFactory : Factory
         private float _heightModifier = 1f;
         [SerializeField]
         private int _sampleCount = 40;
+        [SerializeField]
+        private bool _addMeshCollider = false;
+        [SerializeField]
+        private bool _addToLayer = false;
+        [SerializeField]
+        private int _layerId = 0;
```

Comment:
not all factories actually create game objects (like map image factory). And in mesh factory, it would make much more sense to have layer visualizers have this instead of the factory OR even better, have a layer modifier etc. 
so I'm not sure about moving this down to base class